### PR TITLE
Friendlier error message when process is not found

### PR DESCRIPTION
**Pull request by alonalon**
Give friendlier error message when process is not found as asked in https://github.com/sindresorhus/fkill-cli/issues/30

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -82,3 +82,12 @@ test('ignore ignore-case for pid', async t => {
 
 	await noopProcessKilled(t, pid);
 });
+
+test('error when process is not found', async t => {
+	try {
+		await m(['notFoundProcess']);
+		t.fail();
+	} catch (err) {
+		t.regex(err.message, /Killing process notFoundProcess failed: Process doesn't exist/);
```
Comment (kevva) [inline test.js]: You could use `t.is` here instead since you know the whole error message.

Comment (kevva) [inline test.js]: You could use t.is here instead since you know the whole error message.

**Comment by kevva** [comment]
We should wait for https://github.com/sindresorhus/process-exists/pull/4 to be merged.

**Comment by alonalon** [comment]
Yep, sounds good!

**Comment by kevva** [comment]
The message is `Killing process notFoundProcess failed: Process doesn't exist`. The other stuff is the error stack.

**Comment by alonalon** [comment]
If I do a `console.log('error', err.message);` I got all that as error message.. could it be from aggregate-error module?

**Comment by kevva** [comment]
Yes. I guess you could get the first error from `AggregateError` but I think it's fine as it is, i.e. using `t.regex`.

File: test.js (hunk 1 above)
Comment (sindresorhus) [reply test.js]: Would be better to use `t.throws` here.

File: index.js [hunk 2]
```diff
@@ -43,6 +44,16 @@ function defaultKill(input, opts) {
...
 
+function processExists(proc, list) {
+	let fn = x => x.pid === proc;
+
+	if (typeof proc === 'string') {
+		fn = x => x.name === proc;
+	}
+
+	list.some(x => fn(x));
+}
```
Comment (sindresorhus) [inline index.js]: This function never returns anything.

Comment (alonalon) [reply index.js]: We could pause this mr until we know if https://github.com/sindresorhus/process-exists/pull/4 going to be merged or not. this function should be removed if merged :)

Comment (alonalon) [reply index.js]: Added the new feature from sindresorhus/process-exists#4 and removed this

**Comment by sindresorhus** [comment]
@alonalon Sorry, I merged https://github.com/sindresorhus/fkill/commit/11ad78472b07f7788b1fd7140d2734bea3a06c9c before seeing that this was done. Would you be able to fix the merge conflicts? Or maybe @kevva could help.

**Comment by alonalon** [comment]
No problems, updated. 
Could maybe be little more prettier, but couldn't figure out any good solutions with prettier code.

**Comment by kevva** [comment]
Looks good. I'll do some cleanup in another commit.

**Comment by sindresorhus** [comment]
Thanks @alonalon. Good PR and nice work :)

**Pull request merged**